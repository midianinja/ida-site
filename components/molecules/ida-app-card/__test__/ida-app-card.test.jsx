import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import IDAAppCard from '../ida-app-card';

describe('IDAAppCard', () => {
  afterEach(cleanup);

  it('renders without crashing', async () => {
    const { getByTestId } = render(
      <IDAAppCard
        title="Test"
        imageSrc="/static/icon/ilustra-app-switch.svg"
        url="https://google.com.br"
      />
    );
    expect(getByTestId('image')).toHaveAttribute('src');
  });

  it('matches snapshot card', () => {
    const tree = renderer
      .create(
        <IDAAppCard
          title="Test"
          imageSrc="/static/icon/ilustra-app-switch.svg"
          url="https://google.com.br"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
