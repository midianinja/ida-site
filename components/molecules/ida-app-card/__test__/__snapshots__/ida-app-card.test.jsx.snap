// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDAAppCard matches snapshot card 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <img
    alt="Test"
    data-testid="image"
    src="/static/icon/ilustra-app-switch.svg"
  />
  <h3
    className="title"
  >
    Test
  </h3>
</div>
`;
